server/py/framework/db/sqldb/db.py

```diff
@@ -5929,14 +5930,67 @@ def store_alert_activation(
         self._upsert(session, [alert_activation_record])
 
     def list_alert_activations(
-        self, session: Session, project: typing.Optional[str] = None
+        self,
+        session: Session,
+        projects_with_creation_time: list[tuple[str, datetime]],
+        name: typing.Optional[str] = None,
+        since: typing.Optional[str] = None,
+        until: typing.Optional[str] = None,
+        entity: typing.Optional[str] = None,
+        severity: Optional[
+            list[Union[mlrun.common.schemas.alert.AlertSeverity, str]]
+        ] = None,
+        entity_kind: Optional[
+            Union[mlrun.common.schemas.alert.EventEntityKind, str]
+        ] = None,
+        event_kind: Optional[Union[mlrun.common.schemas.alert.EventKind, str]] = None,
+        page: typing.Optional[int] = None,
+        page_size: typing.Optional[int] = None,
     ) -> list[mlrun.common.schemas.AlertActivation]:
-        # TODO: add filters
         query = self._query(session, AlertActivation)
 
-        if project and project != "*":
-            query = query.filter(AlertActivation.project == project)
+        # Filter alert activations for the project created after the project creation date,
+        # excluding activations linked to any previous instances of the project.
+        # TODO: reconsider this approach when we move alerts out of main MLRun db
+        conditions = [
+            and_(
+                AlertActivation.project == project,
+                AlertActivation.activation_time > created,
+            )
+            for project, created in projects_with_creation_time
+        ]
+
+        query = query.filter(or_(*conditions))
+
+        if name:
+            query = query.filter(
+                generate_query_predicate_for_name(AlertActivation.name, name)
+            )
+
+        if since or until:
+            since = since or datetime.min
+            until = until or datetime.max
+            query = query.filter(
+                and_(
+                    AlertActivation.activation_time >= since,
+                    AlertActivation.activation_time <= until,
+                )
+            )
```

Comment:
this is duplicated in 2 other places (Functions and Artifacts), can be made a common function

Reply:
@TomerShor not sure how it can be generalised when model's and column's names are different

Reply:
Something like this (need to test):
```python
    def _attach_time_range_query(self, query, cls, field_name, since, until):
        since = since or datetime.min
        until = until or datetime.max

        field = getattr(cls, field_name)
        return query.filter(and_(field >= since, field <= until))
```